### ISSUE TITLE
Improve and update SC documentation (Massa docs) (1/3)

### ISSUE THREAD

**Issue by longgamma-massa** [massalabs/massa-docs#7]
### The context
Our [Documentation](https://docs.massa.net/en/latest/index.html) is quite outdated and hard to follow for audience that comes to check and explore Massa's power in practice. Adrien is leading this subject and different Massa departments are working on different segments. 

The objective of Inno team is to focus on the most important new section called 'Web3 developers'. The section includes the following subjects:

- [Smart-contract documentation](https://docs.massa.net/en/latest/web3-dev/smart-contracts.html) - Inno team
- Massa Web3 - Adrien
- [Massa's decentralized web](https://docs.massa.net/en/latest/web3-dev/decentralized-web.html) - Inno team
- Types
- Local network generation
- External resources

**Scope of the task:**
Our target for this task is Smart-contract documentation draft

- introduction
- getting started
- smart contract example (sum)

**The deadline**
1st of November (Tuesday) the draft 
9th of November published finalized Documentation

### The need
The audience who comes to explore Massa should be able to follow SC documentation to start with Massa's smart contracts and follow all examples which demonstrate the capabilities of our SCs.

Put yourself in the shoes of a new dev exploring Massa. How would you impress him with all your knowledge of our SC library?

### How-to
**Content:**
1.  Existing [subsections list](https://docs.massa.net/en/latest/web3-dev/smart-contracts.html) -> update and improve. 
2. Specific focus on SC-examples -> should show the power of the latest version of the massa-sc-library, and provide end-to-end example of a project (for example Token contract or a DApp as in the current version).
3. Add new subsections such as SC-Playground and others as you see necessary.
4. We should cross-link important sections. For example: if we are mentioning Storage costs in the section, cross-link it to the [Storage cost page](https://docs.massa.net/en/latest/technical-doc/storage-costs.html) if it makes sense.

**Where:**
Work directly in the Github Documentation repo (Adrien reviews all work on it).

**Tech details:**

**Comment by qdrn** [comment]
Sounds good to me.

For me the two main points are:
- update documentation so that it uses the new sdk and libraries that we are developing (massa-as-sdk, massa-as, ...). Obviously, making these libraries work is a preriquisite.
- Harmonise all these pages. Right now we have a lot of ressources but it's hard to navigate between things. Like @longgamma-massa said, I think it's important to put ourselves in the shoes of a new dev exploring Massa and help him get setup and have access to the relevant doc. I think we might need a general landing page for the web3 dev section that references all the important information.

**Comment by SlnPons** [comment]
@Thykof if you have time, could you check this Notion page where I drafted the updates do to for the 'web3 developer decentralised web' part please? 
https://www.notion.so/massa-innoteam/Improve-decentralized-web-xThyra-documentation-80481a4e81ea459e997361d4357672f1

Especially the manual way to create DNS, upload website and access the website. I remember there is a new param to add in SCs with testnet15 but I don't have it in mind... 
Cheers

**Comment by Thykof** [comment]
@SlnPons I have created a new issue to split this task: https://github.com/massalabs/massa-docs/issues/13

**Comment by Thykof** [comment]
I have created a new repository for the sum smart contract: https://github.com/massalabs/massa-sc-example-sum
